### PR TITLE
Config webpack listen

### PR DESCRIPTION
**Pull request by normangilmore**
Per #4 discussion, this allows environment variables to set the hostname, listening IP, and port for npm run dev. I also tossed in a title: 'TextThresher' config for  HtmlWebpackPlugin in order to set the page title used by the [template](https://github.com/Goodly/TextThresher/blob/master/app/assets/index_template.html).

Examples:
- `WEBPACK_PORT=3003 npm run dev`
- `WEBPACK_HOSTNAME=192.18.1.100 WEBPACK_LISTEN_IP=0.0.0.0 npm run dev`
- `WEBPACK_HOSTNAME=54.69.118.85 WEBPACK_LISTEN_IP=172.31.45.111 npm run dev`
- `WEBPACK_HOSTNAME=thresher.example.com WEBPACK_LISTEN_IP=0.0.0.0 npm run dev`

0.0.0.0 is not a routable address, just a hacky shorthand for 'all adapters', so the routable address or hostname has to be specified separately.
The hostname alone is insufficient, because very often, the externally reachable IP is not on the local adapter. For example, AWS EC2 instances use a private IP address, and the publicly accessible Elastic IP is assigned externally and lives on some set of Amazon routers and routes to the private IP. So you can't bind to the public IP.

### CODE REVIEW

**Comment by JasmineDeng** [comment]
Does this change how we run it? Do we always now have to specify the `PORT` and `HOSTNAME` variables when we run `npm run dev`?

File: webpack/dev.config.js [hunk 1]
```diff
@@ -5,11 +5,12 @@ import webpack from 'webpack';
...
 
 const PROTOCOL = 'http';
-const HOST = 'localhost';
-const PORT = '3001';
-const PUBLIC_PATH = `${PROTOCOL}://${HOST}:${PORT}/app/`;
+const WEBPACK_LISTEN_IP = process.env.WEBPACK_LISTEN_IP || 'localhost';
+const WEBPACK_HOSTNAME = process.env.WEBPACK_HOSTNAME || 'localhost';
+const WEBPACK_PORT = parseInt(process.env.WEBPACK_PORT, 10) || 3001;
+const PUBLIC_PATH = `${PROTOCOL}://${WEBPACK_HOSTNAME}:${WEBPACK_PORT}/app/`;
+console.log(PUBLIC_PATH);
```
Comment (btburton42) [inline webpack/dev.config.js]: remove console.log

**Comment by btburton42** [comment]
@JasmineDeng no those are optional. defaults to our current options.

**Comment by btburton42** [comment]
@normangilmore nice work. remove log and 👍

**Comment by normangilmore** [comment]
@btburton42 good catch, thanks! removed.
:shipit:

**Pull request merged**